rfc/rfc-0009-supply-chain-switches-template-on-flag.md

```diff
@@ -1,3 +1,197 @@
-[Draft PR](https://github.com/vmware-tanzu/cartographer/pull/75)
+# RFC 0009 Supply Chain Switches Templates
 
-[Draft Text](https://github.com/vmware-tanzu/cartographer/blob/rfc-0009-supply-chain-switches-template-on-flag/rfc/rfc-0009-supply-chain-switches-template-on-flag.md)
+## Summary
+
+Platform creators will need to handle multiple types of workloads. While Cartographer
+supports that behavior currently through allowing multiple supply chains, it should
+reduce the number of supply chains necessary by allowing individual supply chains
+to choose between templates within a step.
+
+## Motivation
+
+A team creating a developer platform on top of Cartographer reached out because
+they maintain supply chains for their platform. They have found that there are
+many small differences in workloads that necessitate making a change to one
+template or another. Currently, such a difference can be handled by:
+
+1. Using ytt within one template to define two different objects.
+2. Creating two templates for to handle the difference and then two supply chains
+   that are the same identical except for one step that uses template A or B.
+
+The team has expressed that they prefer having separate templates (as opposed to
+maintaining ytt templates) but is concerned about the number of supply chains that
+will be needed to pursue this approach. Their hope is to keep the number of
+supply chains small, even as many small differences between workloads could lead to
+a combinatoric explosion of paths to prod.
+
+## Explanation
+
+### Current state of Cartographer
+
+Cartographer is already capable of matching a supply chain to a workload through selectors.
+With no further effort, a platform can be created that has multiple supply chains with different
+purposes or requirements. By specifying characteristics of workloads in the workload labels,
+devs can provide information to operators necessary to choose the right supply chain for
+each workload.
+
+While this behavior is currently possible, relying only on this selection would lead to a platform
+with a large number of supply chains.
+
+Workarounds, for example templates that use ytt templating to embed conditional behavior,
+would represent a code smell that this primitive is not adequate for Cartographer users.
+
+### Condition template choice on information available from the workload
+
+When creating objects of the supply chain, Cartographer has the workload definition available.
+Allowing the supply chain to leverage the workload would add additional power to the conditional.
+
+Each step/resource in a supply chain will continue to define a templateRef, which will continue to
+specify a single template kind (e.g. ClusterImageTemplate). Rather than specifying the name of a
+particular template, authors can provide `options` which is a list of options. An option is
```

Comment:
Is the `options` field mutually exclusive with `name`, or is name removed and the the options list must be used even when there is a single choice?

Reply:
Mutually exclusive. This will not present a breaking change.